src/platform.ml

```diff
@@ -58,9 +58,35 @@ module type THREAD = sig
   (** Re-schedule the calling thread without suspending it. *)
 end
 
+module type RAW_STATS = sig
+  type t = {
+    mutable bytes_read : int;
+    mutable nb_reads : int;
+    mutable bytes_written : int;
+    mutable nb_writes : int;
+  }
+
+  val fresh_stats : unit -> t
+  val reset : t -> unit
+end
+
+module type IO_STATS = sig
+  include RAW_STATS
+
+  val get : unit -> t
+  val get_by_file : string -> t
+  val get_all : unit -> (string * t) list
+  val reset : unit -> unit
+  val bytes_read : unit -> int
+  val bytes_written : unit -> int
+  val nb_reads : unit -> int
+  val nb_writes : unit -> int
+end
```

Comment:
Could you add documentation here to explain the need for 2 separate interfaces?

Reply:
Added some [doc](https://github.com/mirage/index/pull/353/files#diff-f5a28a7f00acf9e80c4d78a180ab01f0cb888ae685b65347a9108d5e4faa420fR62), let me know if its clear.

Reply:
👍